pyiron_base/generic/dynamic.py

```diff
@@ -0,0 +1,60 @@
+import os
+import json
+from pyiron_base.state import state
+
+
+def _get_class_path_lst():
+    path_lst = [
+        os.path.abspath(os.path.expanduser(os.path.join(p, "template")))
+        for p in state.settings.resource_paths
+        if os.path.exists(p) and "template" in os.listdir(p)
```

Comment:
I agree that `template` is not very descriptive and I start to like `templates` more than just `template`. Still I find `scriptjob_templates` too long, as I imagine in future we might have users who only use these template jobs and run `pyiron_base` as a general workflow manager. From their perspective they put their own classes under `pyiron/resources/templates` right next to the configuration of the queuing system in `pyiron/resources/queues`.

Reply:
The reverse argument of course is that the majority of our users might not use `ScriptJobs` and for them it would be confusing to have such a specialized feature living in such a generically named folder.   I don't think folder names can be too long since shells tab-complete, but I also don't care very strongly about this.

Reply:
It might be a bit ambiguous but from my perspective this feature could really help to increase the number of people who just use `pyiron_base` with other all the other functionality. For the users who do not want to use this feature, they do not need to place a `templates` folder in their resource directory, in that case it is less confusing.